lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -354,6 +382,63 @@ public function delete($params = array())
     {
         return $this->getService()->getClient()->delete($this->getUrl())->send();
     }
+    
+    /**
+     * @param string $destination Path (`container/object') of other object to symlink this object to
+     * @return null|\Guzzle\Http\Message\Response The response or null if $this is not empty
```

Comment:
Instead of returning `null` for an error case, lets throw an exception instead. This has the following benefits:
* it will keep error-handling consistent within this method. Just as you throw `Exceptions\NoNameError` when the object has no name, you can throw another exception if the object has content in it.
* by using an exception instead of returning `null`, we can provide more information to the caller about what went wrong.

Feel free to re-use an exception class from https://github.com/markchalloner/php-opencloud/tree/working/lib/OpenCloud/Common/Exceptions or add your own if you can't find one that's appropriate.